include/micm/solver/solver.hpp

```diff
@@ -40,13 +40,13 @@ namespace micm
     Solver& operator=(const Solver&) = delete;
 
     Solver(Solver&& other)
-        : number_of_grid_cells_(other.number_of_grid_cells_),
+        : solver_(std::move(other.solver_)),
+          processes_(std::move(other.processes_)),
+          number_of_grid_cells_(other.number_of_grid_cells_),
           number_of_species_(other.number_of_species_),
           number_of_reactions_(other.number_of_reactions_),
           state_parameters_(other.state_parameters_)
     {
-      std::swap(this->solver_, other.solver_);
-      std::swap(this->processes_, other.processes_);
```

Comment:
@K20shores @mattldawson I thought doing `swap` here would make sure when `other` is destroyed by the move constructor, it did not free the space now owned by `this`. Kyle's change basically means this is not necessary?

Reply:
This `Solver(&&)` move constructor caused an error in musica, and I'm working on it. Also the move assignment has a bug in it. We might have to add test for this.  

```c++
    Solver& operator=(Solver&& other)
    {
      std::swap(this->solver_, other.solver_);
      number_of_grid_cells_ = other.number_of_grid_cells_;
      number_of_species_ = other.number_of_species_;
      number_of_reactions_ = other.number_of_reactions_;
      state_parameters_ = other.state_parameters_;
      std::swap(this->solver_, other.processes_); <---- this should get corrected
      return *this;
    }
```

Reply:
no problem! yeah I don't see the need to have move assignment.